src/dodal/devices/xspress3/xspress3.py

```diff
@@ -0,0 +1,144 @@
+
+    Parameters
+    ----------
+    prefix:
+        Beamline part of PV
+    name:
+        Name of the device
+    num_channels:
+        Number of channel xspress3 has, default is 1 for mini.
+    timeout:
+        How long to wait for before timing out for staging/arming of detector default is 1 sec
+    """
+
+    def __init__(
+        self, prefix: str, name: str = "", num_channels: int = 1, timeout=1
+    ) -> None:
+        self.channels = DeviceVector(
+            {i: Xspress3Channel(f"{prefix}C{i}_") for i in range(1, num_channels + 1)}
+        )
+        """MCA on/off switch readback"""
+        self.get_roi_calc_status = DeviceVector(
+            {
+                i: epics_signal_rw(float, f"{prefix}MCA{i}:Enable_RBV")
+                for i in range(1, num_channels + 1)
+            }
+        )
+        """start and size of the multi-channel analyzer (MCA) array"""
+        self.roi_mca = DeviceVector(
+            {
+                i: Xspress3ROIChannel(f"{prefix}ROISUM{i}:")
+                for i in range(1, num_channels + 1)
+            }
+        )
+
+        """signal for the corrected MCA spectrum (1d array)"""
+        self.dt_corrected_latest_mca = DeviceVector(
+            {
+                i: epics_signal_r(NDArray[float64], f"{prefix}ARR{i}:ArrayData")
+                for i in range(1, num_channels + 1)
+            }
+        )
+
+        """Shared controls for triggering detection"""
+        self.timeout = timeout
+        self.acquire_time = epics_signal_rw(float, prefix + "AcquireTime")
+        self.max_num_channels = epics_signal_r(int, prefix + "MAX_NUM_CHANNELS_RBV")
+        # acquire and acquire readback has a different enum
+        self.acquire = epics_signal_rw(AcquireState, prefix + "Acquire")
+        self.acquire_rbv = epics_signal_r(AcquireRBVState, prefix + "Acquire_RBV")
+        self.trigger_mode = epics_signal_rw_rbv(TriggerMode, prefix + "TriggerMode")
+
+        self.detector_state = epics_signal_r(
+            DetectorState, prefix + "DetectorState_RBV"
+        )
+
+        self.set_num_images = epics_signal_rw(int, prefix + "NumImages")
+        self.detector_busy_states = [
+            DetectorState.ACQUIRE,
+            DetectorState.CORRECT,
+            DetectorState.ABORTING,
+        ]
+        super().__init__(name=name)
+
+    @AsyncStatus.wrap
+    async def stage(self) -> None:
+        LOGGER.info("Arming Xspress3 detector...")
+        await self.trigger_mode.set(TriggerMode.BURST)
+        await wait_for_value(
+            self.detector_state,
+            lambda v: v in self.detector_busy_states,
+            timeout=self.timeout,
+        )
+        await self.acquire.set(AcquireState.ACQUIRE)
+        await wait_for_value(
+            self.acquire_rbv, AcquireRBVState.ACQUIRE, timeout=self.timeout
+        )
+        await wait_for_value(
+            self.acquire_rbv, AcquireRBVState.DONE, timeout=self.timeout
+        )
```

Comment:
Should: I don't think it makes sense to wait on both of these `stage` basically means get the device ready to take data so I expect that we want to wait for `ACQUIRE` (it is now taking data) rather than `DONE` (it has finished taking data).

Reply:
I think I got confused with the return await_value(self.acquire, 0). 
```python 
acquire = Component(EpicsSignalWithRBV, "Acquire")

def arm(self) -> Status:
        LOGGER.info("Arming Xspress3Mini detector...")
        self.trigger_mode_mini.put(TriggerMode.BURST.value)
        arm_status = await_value_in_list(self.detector_state, self.detector_busy_states)
        self.erase.put(EraseState.ERASE.value)
        arm_status &= self.acquire.set(AcquireState.ACQUIRE.value)
        arm_status.wait(self.ARM_STATUS_WAIT)
        return await_value(self.acquire, 0)
 ```

would this be more correct?
 ```python
    @AsyncStatus.wrap
    async def stage(self) -> None:
        LOGGER.info("Arming Xspress3 detector...")
        await self.trigger_mode.set(TriggerMode.BURST)
        await wait_for_value(
            self.detector_state,
            lambda v: v in self.detector_busy_states,
            timeout=self.timeout,
        )
        await self.acquire.set(AcquireState.ACQUIRE)
        await wait_for_value(
            self.acquire_rbv, AcquireRBVState.ACQUIRE, timeout=self.timeout
        )
    @AsyncStatus.wrap
    async def unstage(self) -> None:
        LOGGER.info("unstaging Xspress3 detector...")
        await wait_for_value(
            self.acquire_rbv, AcquireRBVState.DONE, timeout=self.timeout
        )
 ```

Reply:
I think the waiting for the done in `unstage` makes more logically sense to me as something we would expect on the device. So let's leave it at that and make sure it works as expected as part of https://github.com/DiamondLightSource/hyperion/issues/1433